### PR TITLE
[Azure Logs] Expand date formats for parsing time fields

### LINKED ISSUE #15977: Update `time` format across all Azure Logs integrations
## Context

Log events from Azure usually respect the spec, and the `time` field comes as an ISO8601 string.

Unfortunately, some services sometimes use a different format. For example, the `time` value is sometimes `11/14/2025 1:48:53 AM` instead of `2025-11-14T01:48:53.000Z`, and the pipeline fails.

We recently updated the signin logs integration with https://github.com/elastic/integrations/pull/15976, but should review all Azure integrations to double-check all integrations have a fallback list of date formats.

## Goal

- Review all existing integrations looking for date formats in use.
- Update the integrations with a list of existing formats, ordered from most to least frequently used.

## Impact

Users may experience pipeline failures when an Azure service ships log events with off-spec values.

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

We are revising the date processor format list to improve robustness. We've prioritized the formats based on usage frequency: the first format is the most common, followed by necessary fallbacks.

This addresses the fact that Azure Logs don't adhere to a single format. This approach maximizes efficiency while maintaining parsing resilience.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] ~~I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)~~

## How to test this PR locally

```sh
cd packages/azure
elastic-package build && elastic-package stack up -d -v --version 9.2.1
elastic-package test pipeline -v
```

## Related issues

- Closes #15977

### CODE REVIEW

File: packages/azure/data_stream/identity_protection/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -37,9 +37,16 @@ processors:
       field: azure.identityprotection.time
       target_field: '@timestamp'
       ignore_failure: true
-      formats:
+      # Order is prioritized by observed frequency for this log
+      # category.
+      # We expect the first format to be used in 99% of cases.
+      # However, we include the other formats seen in other
+      # categories as a necessary fallback.
+      formats:
```
Comment (muthu-mps) [inline packages/azure/data_stream/identity_protection/elasticsearch/ingest_pipeline/default.yml:45]: Should we add this format as well?

`- "M/d/yyyy h:mm:ss a XXX"`

Comment (zmoog) [reply packages/azure/data_stream/identity_protection/elasticsearch/ingest_pipeline/default.yml:45]: Done!

File: packages/azure/changelog.yml [hunk 2]
```diff
@@ -1,3 +1,8 @@
+- version: "1.30.1"
+  changes:
+    - description: Expand date formats for parsing time fields.
```
Comment (StacieClark-Elastic) [inline packages/azure/changelog.yml]: Please list the added formats in the changelog so customers can see what was changed.

**Comment by StacieClark-Elastic** [comment]
For a bug fix, I would expect to see some tests that verify that the bug has been fixed. There are no changes to the tests so running the test suite only validates that the old tests did not get broken.

**Review by kaiyan-sheng — APPROVED** [review]
Maybe we can add a sample log for each additional format as @StacieClark-Elastic mentioned about test?

**Comment by zmoog** [comment]
> For a bug fix, I would expect to see some tests that verify that the bug has been fixed. There are no changes to the tests so running the test suite only validates that the old tests did not get broken.

I agree, it makes sense to add more tests to prove the change is effective. 

To add more context, this is not addressing a user-reported actual occurrence of this problem. This is mostly a defense against future off-spec changes, so we don't know which data stream will receive off-spec dates yet. Azure often emits off-spec values in dates and all sort of other fields.

I classified this PR as a bugfix because Azure-focused integrations have to work in this environment.

**Comment by zmoog** [comment]
> Maybe we can add a sample log for each additional format as @StacieClark-Elastic mentioned about test?

Yeah, and I guess we need to do it for every data stream in the Azure Logs package.

**Comment by giorgi-imerlishvili-elastic** [comment]
Tests related to the changes would be nice. Otherwise, LGTM!

**Comment by zmoog** [comment]
Thanks for all the comments and suggestions: collecting time format samples to add a "corner cases time formats" test file for each data stream.

**Comment by zmoog** [comment]
@StacieClark-Elastic, added the following test case with the know time formats:

```json
{"time": "01/09/2007 09:41:00", "category": "Administrative"}
{"time": "1/9/2007 09:41:00", "category": "Administrative"}
{"time": "01/09/2007 09:41:00 AM", "category": "Administrative"}
{"time": "1/9/2007 9:41:00 AM", "category": "Administrative"}
{"time": "1/9/2007 10:41:00 AM +01:00", "category": "Administrative"}
{"time": "2007-01-09T09:41:00", "category": "Administrative"}
{"time": "2007-01-09T09:41:00.22Z", "category": "Administrative"}
{"time": "2007-01-09T09:41:00.6816663Z", "category": "Administrative"}
{"time": "2007-01-09T09:41:00.535404056Z", "category": "Administrative"}
{"time": "2007-01-09T09:41:00.992099+00:00", "category": "Administrative"}
{"time": "2007-01-09T11:41:00+02:00", "category": "Administrative"}
```

Then I added the test case to all data streams that attempt to parse the `@timestamp` field.

This should ensure all data streams support the known data format variations we spotted in the wild. Please let me know what you think of this approach.

File: packages/azure/changelog.yml (hunk 2 above)
Comment (zmoog) [reply packages/azure/changelog.yml]: Done!

**Comment by zmoog** [comment]
> For a bug fix, I would expect to see some tests that verify that the bug has been fixed. There are no changes to the tests so running the test suite only validates that the old tests did not get broken.

@StacieClark-Elastic, please let me know if you think the change requires more/different tests coverage.

**Comment by zmoog** [comment]
@StacieClark-Elastic, I'd appreciate your review when you get a moment! 🙇
